.github/workflows/ci.yml

```
@@ -18,8 +18,8 @@ jobs:
         uses: actions/checkout@v4
       - name: Run ShellCheck bin top level
         run: |
-            shellcheck bin/support/bash_functions.sh bin/support/download_ruby -x &&
-            shellcheck bin/compile bin/detect bin/release bin/test bin/test-compile -x
+            git ls-files --cached --others --exclude-standard ':!:*.rb' 'bin/*' '*/bin/*' '*.sh' | \
+              xargs shellcheck --check-sourced --color=always
```

TIL about the [`pathspec` syntax of Git](https://git-scm.com/docs/gitglossary#Documentation/gitglossary.txt-aiddefpathspecapathspec). That took some digging, so proposing a docs link ;)

Also, zero-byte delimiters for `xargs` are always a good idea.

```suggestion
            # https://git-scm.com/docs/gitglossary#Documentation/gitglossary.txt-aiddefpathspecapathspec
            git ls-files -z --cached --others --exclude-standard ':(exclude):*.rb' 'bin/*' '*/bin/*' '*.sh' | \
              xargs -0 shellcheck --check-sourced --color=always
```

---

Updated in 

- https://github.com/heroku/heroku-buildpack-ruby/pull/1635/commits/3e776106a4d29aeb7cba63eff83fc026ee93ef36
- https://github.com/heroku/heroku-buildpack-ruby/pull/1635/commits/c2422e0ccdff574a7fd77b32c89bb9a05bf2b287